src/dorsal_deferred.js

```diff
@@ -88,6 +89,13 @@ DorsalDeferred = function(instances) {
     dfd.promise = function() {
         return promise;
     };
+    dfd.when = function(condition) {
```

Comment:
I think `when` should only accept an array of promises (or deferreds), and return a new promise.